### ISSUE TITLE
quicktest test_email_with_quoted_printable_body fail

### ISSUE THREAD

**Issue by brucegibbins** [django-helpdesk/django-helpdesk#1079]
Hi. This is possibly more of a question than an issue.

Platform: Windows 11
Locale: Australia
IDE: PyCharm

I am trying to prep my dev environment so that I can run tests and potentially contribute but I am getting an issue when running quicktest. Basically the test fails at line 82 with the following assertion. The reason for the failure is that ticket.description has a trailing '\n' (linefeed) and therefore does not match the assertion test. 

```
        self.assertEqual(ticket.description,
                         "Tohle je test českých písmen odeslaných z gmailu.")
```

Tracing the code leads me to the extract_part_data() function that does a number of  things but after line 749 (below) part_body variable contains the string with the trailing linefeed after extracting the part from the message object. 

`part_body = part.get_payload(decode=True)
`

I am just wondering if this related to some type of Windows Specific management of the carriage return line feed or something to do with utf-decoding which i am not familiar with?

I have tested this within PyCharm so that I could trace line execution but get the same issue from the CLI as well.

Thank-you in advance.

**Comment by timthelion** [comment]
The text says, 'this is a test of czech characters sent from gmail. Nothing to do with linefeeds. I think that the testis still perfectly valid if you just strip the description when passing it to asstert.

**Comment by brucegibbins** [comment]
Hi. Thanks. Yes I agree that the test is still valid. I wasn't actually
concerned about the text itself which in the assertion test does not
include a linefeed. But when the description field is extracted, at least
in my case, it includes a single trailing linefeed which it probably
collected from the test message. Therefore when the comparison occurs they
are identical except for the inclusion of the trailing linefeed after the
message body is extracted.

So I think the test is more than valid. But am trying to isolate why in my
case all of the trailing linefeeds were dropped except for one?

Sorry for the confusion.

**Comment by brucegibbins** [comment]
I have attached a couple of screenshots to show what I mean...

The first image shows the end-of-line characters (in my case \r\n)

![eml](https://user-images.githubusercontent.com/16873240/226559023-0a5f7bd0-28bf-4573-a661-42295b74902e.png)

The second image is what is shown during debug. As you can see there is a trailing linefeed which on the face of it is not a problem except that the test expects that there are no trailing end of line markers.  Which again is logical. But of course the test will fail in this case.

![email](https://user-images.githubusercontent.com/16873240/226560008-1cd96c8f-6a47-4dd4-b56d-da8630903a03.png)

**Comment by uhurusurfa** [comment]
This is probably caused because he file is being read using your system default which is not UTF-8.
Add encoding="utf-8" to all of the open() call in the test_get_email.py file.

**Comment by brucegibbins** [comment]
Ok. Thanks

I have looked into changing Windows System Default and can see that Windows
11 has a BETA region locale setting to enable UTF-8. But since this may
have some other downstream impact I will trial this at some other time and
in the short term add the utf-8 encoding parameter.

**Comment by brucegibbins** [comment]
Sorry to keep bothering you. I have added encoding='utf'8' to all open
context calls and have also enabled a beta utf8 encoding windows 11 region
locale setting.

Still results in this single trailing linefeed.

Can you tell me... the original .eml files used for testing do they have
just linefeeds or carriage return linefeeds?

Mine have carriage return linefeeds and I am wondering if they have been
altered somewhere to the windows default.

This should still not break the tests but I am just curious but will
continue to investigate.

Cheers

**Comment by uhurusurfa** [comment]
The source eml file has only ever been committed once.

Check you have not modified the file being read that appears to be causing your problem:
`git status helpdesk/tests/test_files/quoted_printable.eml`

To verify that you have not somehow inadvertently commited a change, run this from your command line in the project base directory:
`git log helpdesk/tests/test_files/quoted_printable.eml`

You should see this:
```
django-helpdesk % git log helpdesk/tests/test_files/quoted_printable.eml
commit 2c3ce8903fed7b15ecfabc937cbad1a959db1c65
Author: Timothy Hobbs <timothy@hobbs.cz>
Date:   Wed Mar 6 13:49:23 2019 +0100

    Decode quoted-printable html bodies when getting email

    Fixes #719
django-helpdesk %
```

**Comment by brucegibbins** [comment]
Thank. Yes I expect I have triggered something as I had started ro look at
an issue we are having with the lost off some of then content from emailed
tickets.

I an now able to run the tests beyond this point so now I need to establish
why the small chamge I had affected the rendering of the nessage parts.

Thank you again for your support.
